File: cloud-reactor-api/src/main/java/com/sequenceiq/cloudbreak/cloud/event/resource/TerminateStackResult.java
package com.sequenceiq.cloudbreak.cloud.event.resource;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.sequenceiq.cloudbreak.cloud.event.CloudPlatformResult;
import com.sequenceiq.cloudbreak.common.event.FlowPayload;

public class TerminateStackResult extends CloudPlatformResult implements FlowPayload {

    public TerminateStackResult(Long resourceId) {
        super(resourceId);
    }

    @JsonCreator
    public TerminateStackResult(
            @JsonProperty("statusReason") String statusReason,
            @JsonProperty("errorDetails") Exception errorDetails,
            @JsonProperty("resourceId") Long resourceId) {
        super(statusReason, errorDetails, resourceId);
    }
}

File: cloud-reactor-api/src/main/java/com/sequenceiq/cloudbreak/cloud/event/resource/TerminateStackResult.kt
package com.sequenceiq.cloudbreak.cloud.event.resource

import com.sequenceiq.cloudbreak.cloud.event.CloudPlatformResult

class TerminateStackResult : CloudPlatformResult<TerminateStackRequest<TerminateStackResult>> {

    constructor(request: TerminateStackRequest<TerminateStackResult>) : super(request) {
    }

    constructor(statusReason: String, errorDetails: Exception, request: TerminateStackRequest<TerminateStackResult>) : super(statusReason, errorDetails, request) {
    }
}
